### PR TITLE
Fix FrameworkInfo::user documentation

### PR DESCRIPTION
**Pull request by ktf**
Add a note about the user being required in case the HTTP Scheduler API
is used.

### CODE REVIEW

**Comment by haosdent** [comment]
Please update `include/mesos/mesos.proto` as well.

**Comment by ktf** [comment]
Done.

**Comment by haosdent** [comment]
LGTM. :-)

File: include/mesos/mesos.proto [hunk 1]
```diff
@@ -205,9 +205,12 @@ message MachineInfo {
...
 message FrameworkInfo {
-  // Used to determine the Unix user that an executor or task should
-  // be launched as. If the user field is set to an empty string Mesos
-  // will automagically set it to the current user.
+  // Used to determine the Unix user that an executor or task should be
+  // launched as. If the user field is set to an empty string
+  // MesosSchedulerDriver will automagically set it to the current user.
+  //
+  // NOTE: The user has to be set explicitly in case the HTTP API
+  // is used.
```
Comment (zhitaoli) [inline include/mesos/mesos.proto]: Can we restructure these two sentences as this is not very clear.

Maybe consider something among:

```
// Used to determine the Unix user that an executor or task should be
// launched as.
//
// If using HTTP API, this field must be explicitly set.
//
// If using MesosSchedulerDriver and this field is set to an empty string, MesosSchedulerDriver
// will automagically set it to the current user.

**Comment by ktf** [comment]
Hopefully improved.

**Comment by Gilbert88** [comment]
Thanks for the clarification @ktf .

This patch LGTM!

**Pull request closed without merging**